src/main/java/lotto/domain/NumberElement.java

```diff
@@ -0,0 +1,46 @@
+package lotto.domain;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NumberElement {
+
+    private static final Map<Integer, NumberElement> numberElementMap = new HashMap<>();
+
+    static {
+        for (int i = 1; i <= 45; i++) {
+            numberElementMap.put(i, new NumberElement(i));
+        }
+    }
+
+    private final int value;
+
+    public NumberElement() {
+        this(0);
+    }
+
+    public NumberElement(int value) {
```

Comment:
```suggestion
    private NumberElement(int value) {
```
기본 생성자 제거
외부에서 객체를 생성할 수 없도록 private으로 구현하는 것은 어떨까?
정적 팩터링 메서드를 제공하는 경우 private으로 구현하는 것을 추천